server/mongo_indexes.go

```diff
@@ -0,0 +1,199 @@
+package server
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"strings"
+
+	mgo "gopkg.in/mgo.v2"
+)
+
+// IndexMap is a map of index arrays with the collection name as the key. Each index array
+// contains one or more *mgo.Index indexes.
+type IndexMap map[string][]*mgo.Index
+
+// ConfigureIndexes ensures that all indexes listed in the provided indexes.conf file
+// are part of the Mongodb fhir database. If an index does not exist yet ConfigureIndexes
+// creates a new index in the background using mgo.collection.EnsureIndex(). Depending
+// on the size of the collection it may take some time before the index is created.
+// This will block the current thread until the indexing completes, but will not block
+// other connections to the mongo database.
+func ConfigureIndexes(session *mgo.Session, config Config) {
+	var err error
+
+	db := session.DB(config.DatabaseName)
+
+	// Read the config file
+	idxConfig, err := ioutil.ReadFile(config.IndexConfigPath)
+	if err != nil {
+		log.Println("[WARNING] Could not find indexes configuration file")
+	}
+
+	// parse the config file
+	indexMap, err := parseIndexes(string(idxConfig))
+
+	if err != nil {
+		log.Fatal(err.Error())
+	}
+
+	// ensure all indexes in the config file
+	for k := range indexMap {
+		collection := db.C(k)
+
+		for _, index := range indexMap[k] {
+			log.Printf("Ensuring index: %s.%s: %s\n", config.DatabaseName, k, sprintIndexKeys(index))
+			err = collection.EnsureIndex(*index)
+
+			if err != nil {
+				log.Printf("[WARNING] Could not ensure index: %s.%s: %s\n", config.DatabaseName, k, sprintIndexKeys(index))
+			}
+		}
+	}
+	session.Close()
+}
+
+func parseIndexes(fileContents string) (IndexMap, error) {
+	var indexMap = make(IndexMap)
+	lines := strings.Split(fileContents, "\n")
```

Comment:
This approach is fine here (the file's not _too_ big), but the idiomatic way to read a file line-by-line in Go is to use a `bufio.Scanner`.  Here's an example from one of my other projects:
https://github.com/intervention-engine/integrator/blob/master/main.go#L142-L158